### ISSUE TITLE
Keg Fill Image Displaying Inverse of Level

### ISSUE THREAD

**Issue by thadius856** [RaspberryPints/RaspberryPints#17]
During the CSV->SQL conversion/CSS cleanup, some image updates broke.

In particular, the keg graphic is showing the inverse level of what it should be. For example, a keg down 5 pints should show approximately 90% full (10% empty). Instead, it is showing 10% full (90% empty). See screenshot:

![untitled](https://f.cloud.github.com/assets/6320180/1866573/58daf214-7836-11e3-9ec0-b51ea8f0b998.png)

**Comment by JayUnt** [comment]
we might want to hold off on fixing this until the branches have been merged into the trunk.  But i can look into it when i do the merging.

**Comment by JayUnt** [comment]
it looks as if the merge removed all of the css image changes that i made.  I can re-integrate my stuff, but thats annoying that it would do that.  The current codebase using the original 10.png, 20.png, etc.

**Comment by thadius856** [comment]
Ugh. Stupid thing. Sorry. :(

I think GitHub lets you go back and see previous commits.

I also have old commits backed up on my HDD (commits #10/11/13/17/31/34). Let me know if you want me to send you any.

**Comment by Skemp45** [comment]
can you also paste in the old code from the csv file for this section?

**Comment by JayUnt** [comment]
I have the code from the fork I created, so it wont take much.  I'll do it tonight unless Skemp did it already.  He's plowing through all of this.

**Comment by Skemp45** [comment]
I have not touched the Keg stuff I will let you attack that!!!!

**Comment by JayUnt** [comment]
Do either of you know how i can replace my fork with the master? I cant find it on the site, and the closest i can see is to delete my fork and create a new one.

**Comment by JayUnt** [comment]
nevermind, i think i got it. i had to use the git command line, which i havent used before.

**Comment by JayUnt** [comment]
I started to work on this but I'm hitting a snag with this css style.

.BeerWrapper div{
    min-height:50px;
    height:auto !important;
    height:50px;
    color: white;
    float: left;
}

I need to set a height on some divs within the BeerWrapper class, but its being overwritten by the the "height: auto !important" of this style.  This is one of the reasons i dont like using floating divs to mimic a table structure.  Since you dont want to use a table, do you mind if i re-write this to use inline-blocks?  I think its a happy medium.

What do you think?

**Comment by thadius856** [comment]
Of course the height is being overwritten by !important. Why are you surprised? You defined height twice in that snippet.

So you want the BeerWrapper to be auto, but some of the BeerWrapper child divs to be 50px?

```
<div class="BeerWrapper">
     <div>Stuff 1</div>
     <div>Stuff 2</div>
     <div>Stuff 3</div>
</div>
```

```
.BeerWrapper { height: auto; }
.BeerWrapper div { height: 50px; }
```

**Comment by thesehands** [comment]
Its is a box model issue that solved cross browser issues with auto heights. I will fix the child div issue tomorrow

**Comment by thadius856** [comment]
Searched for 30 minutes before I found the words for what I was describing above. This method is called Descendant Selectors.

When I specified `.BeerWrapper div { height: 50px; }`, what I was describing, in plain English, was this:
- If a div is nested anywhere within an object of the BeerWrapper class, make it a height of 50px.

or, the other way around,
- Apply a height of 50 px to any div nested within an object of the BeerWrapper class.

This applies even if I nest deeper. For example:

`<div class="BeerWrapper"><div><table><tr><td><div></div></td></tr></table></div>`

The outermost div will be `height: auto`, while the middle div will get `height: 50px`. The `table`, `tr`, `td` will get nothing, and the inner-most div will also get `height: 50px`.

It would probably be better to use Child Selectors, so it only cascades one level of nesting deep.

`.BeerWrapper { height: auto; }
.BeerWrapper > div { height: 50px; }`

In this situation, only the middle div would get `height: 50px`. The inner-most div gets nothing.

I don't think you need `!important`, as the decendant/child is evaluated after the parent, so it would overwrite it. Either way, I suppose it couldn't hurt.

**Comment by JayUnt** [comment]
Yeah, having !important in the css just wreaks havoc.  Once you have it in there, then you need to add more of them to fix other spots.

I updated my fork and to fix this keg issue as well as updated the css to fix the !imporant stuff.  But now im having an issue committing it back into my fork.  Ill let you know when its in there.

**Comment by thadius856** [comment]
Agreed.

Yeah, the forking/branching and merging back is a pretty big headache of mine.

So far, about 50% of the pull requests have been able to merge back flawlessly. Others, not so much. I haven't been able to discern why yet. :\

**Comment by JayUnt** [comment]
So i found out i can use svn git this repository, so i pulled it down again using that, updated my code and committed it. :score:  i just sent a pull request, but it looks like this is going to be one of those "not so much" ones.  sorry about that.

**Comment by thadius856** [comment]
Wow, I'm see breakage in a major way.

![untitled](https://f.cloud.github.com/assets/6320180/1884605/a9c8432c-79a6-11e3-856c-ac3abe372b68.png)

What... happened?

**Comment by thadius856** [comment]
On closer look, I see a couple odd lines added in.

`<<<<<<< HEAD`

`>>>>>>> CSS, SQL Change`

This document has two bodies and the Tap # column disappeared from the code entirely.

**Comment by Skemp45** [comment]
Oh my what happened!!!

**Comment by JayUnt** [comment]
Doh! I guess the fork stuff didn't work like I thought it did. Let me see what I can do.

**Comment by JayUnt** [comment]
well that sucks. everything i worked on is lost.  ill work on it again tomorrow and hopefully it wont be lost this time,

**Comment by thadius856** [comment]
:o sry

Least you remember what you did? :)

**Comment by JayUnt** [comment]
Ok i committed again. This time i deleted my repo and make a brand new fork.  I am now also using svn to pull it down, which I'm more comfortable with.  I just requested a pull and hopefully it will so smoothly since its small changes.

**Comment by Skemp45** [comment]
nice, the keg works great I did however find a couple issues, I did not have much time to look into it but here it is. 
![error](https://f.cloud.github.com/assets/6318869/1891451/67a992d8-7a3d-11e3-9d54-362c9c8a16c3.png)

**Comment by Skemp45** [comment]
if they have 0 left in a keg we should be able to display a empty message or something. 
and I have no idea why the off the charts cup is displaying over there

**Comment by JayUnt** [comment]
good catch.  ill work on that tonight. it wont take long.

On Fri, Jan 10, 2014 at 4:24 PM, Skemp45 notifications@github.com wrote:

> if they have 0 left in a keg we should be able to display a empty message
> or something.
> and I have no idea why the off the charts cup is displaying over there
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/thadius856/Digital-Taplist/issues/17#issuecomment-32067583
> .

## 

Jason Unterman

**Comment by Skemp45** [comment]
also, whats causing the borders to have a gap again? Any ideas

**Comment by JayUnt** [comment]
I fixed the abv image issue and the keg divide by 0 issue directly on the master.  I also made sure the ibu is centered.  I had it centered on my screen, but when you stretch the screen it wasnt anymore.  But its now working right.  

I will look into the gaps tonight.

**Comment by Skemp45** [comment]
Hell yeah man nice work.

**Comment by JayUnt** [comment]
Hey guys,  Please take a look at the code in my fork. I updated it to use a table for the beer rows.  The files are a bit smaller and I personally think the code is a lot cleaner.  It also fixed the border bug that we were having.

Below is a screenshot so you can see what it looks like if you dont want to pull it down.  Let me know what you guys think.

![2014-01-10 22_42_04-kegerface v0 5](https://f.cloud.github.com/assets/35524/1893152/89528158-7a72-11e3-9512-ae46e15d3db0.png)

**Comment by thadius856** [comment]
_glares_ Hrmmm... well..

It certainly looks nice (damn you).

Will we still be able to turn a single column on/off? I'm guessing it's still possible by putting an IF before the`td` in the header row and before the corresponding `td` in the body of the table.

Does this make v-align any easier?

Edit: Does auto-width still work?

**Comment by thadius856** [comment]
Fuck it. I'm sold.

Here we go. Hope you're ready to support any oddities that result from tables.

Merging branch momentarily.

**Comment by JayUnt** [comment]
Yeah, turning off a column would just be a matter of not printing the column.  Ive never tried to hide a td using display:none, but that might be a possibility too.  

that tap num is vertical aligned automagically (as git likes to call it), but if we wanted it top aligned all we would have to do is add "vertical-align: top" to that class.  So yeah, its pretty easy.

And right now all of the columns have a width percentage set, except the beer name.  So that one will expand/collapse when you stretch/shrink the window.  Until it gets too narrow, in which case it stops shrinking.

**Comment by thadius856** [comment]
Gah. GH is being gay again and refuses to merge.

Can you make this happen, or shall I manual merge?

**Comment by JayUnt** [comment]
I will copy the files to master and commit.

**Comment by JayUnt** [comment]
done and done.

**Comment by thadius856** [comment]
Thanks!
